acceptance/openstack/networking/v2/extensions/agents/agents_test.go

```diff
@@ -92,3 +95,84 @@ func TestAgentsRUD(t *testing.T) {
 	err = agents.Delete(client, allAgents[0].ID).ExtractErr()
 	th.AssertNoErr(t, err)
 }
+
+func TestBGPAgentRUD(t *testing.T) {
+	waitTime := 30 * time.Second
```

Comment:
There's a [WaitFor() helper function](https://github.com/gophercloud/gophercloud/blob/0ab2b5dec6f8e9f6b084f99c908996b6493d3c09/acceptance/tools/tools.go#L17) if you need to wait for a certain condition to be satisfied. You can use it instead of the time.Sleep() calls you've added to this test.

Reply:
This is really tricky. `waitFor` can only wait 10 minutes, that's too much. On the other hand, `waitForTimeout` might be better, however this is not a timeout problem.